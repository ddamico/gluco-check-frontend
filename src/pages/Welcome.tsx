import React from "react";
import { useTranslation } from "react-i18next";
import {
  Button,
  Container,
  Grid,
  makeStyles,
  Typography,
} from "@material-ui/core";
import Onboarding from "../components/Onboarding";
import Boilerplate from "../components/Boilerplate";

const useStyles = makeStyles((theme) => ({
  container: {
    paddingTop: theme.spacing(2),
    textAlign: "center",
  },
  subtitle: {
    marginBottom: theme.spacing(2),
    textAlign: "center",
  },
  bodyContent: {
    "& p": {
      marginBottom: theme.spacing(2),
    },
    "& ul li": {
      listStyle: "none",
      marginBottom: theme.spacing(1),
      fontWeight: 600,
    },
  },
  contentGrid: {
    minHeight: "90vh",
  },
  boilerplate: {
    textAlign: "center",
    marginTop: theme.spacing(2),
    "& p": {
      marginBottom: theme.spacing(1),
      [theme.breakpoints.down("sm")]: {
        fontSize: ".7rem",
      },
    },
  },
  ctaContainer: {
    "&.MuiGrid-item": {
      padding: "0",
    },
  },
}));

function Welcome() {
  const classes = useStyles();
  const { t } = useTranslation();

  return (
    <Grid
      className={classes.contentGrid}
      container
      direction="column"
      justify="center"
      alignItems="center"
      spacing={2}
      wrap="nowrap"
    >
      <Grid item className={classes.subtitle}>
        <Container maxWidth="sm">
          <Typography variant="h5" component="h2">
            {t("welcome.subtitle")}
          </Typography>
        </Container>
      </Grid>
      <Grid item>
        <Onboarding />
      </Grid>
      <Grid item>
        <Button variant="contained" color="primary" href="/settings">
          {t("welcome.cta")}
        </Button>
      </Grid>
      <Grid item className={classes.boilerplate}>
        <Boilerplate />
      </Grid>
    </Grid>
  );
}

export default Welcome;
