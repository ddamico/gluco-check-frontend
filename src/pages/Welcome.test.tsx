import React from "react";
import { cleanup, render } from "@testing-library/react";
import { axe } from "jest-axe";
import Welcome from "./Welcome";

jest.mock("react-i18next", () => ({
  useTranslation: () => {
    return {
      t: jest.fn().mockImplementation((i) => {
        return i;
      }),
    };
  },
  Trans: function (props: any) {
    return <span>{props.i18nKey}</span>;
  },
}));

jest.mock("../components/Onboarding", () => {
  return {
    __esModule: true,
    default: () => {
      return <>Onboarding</>;
    },
  };
});

jest.mock("../components/Boilerplate", () => {
  return {
    __esModule: true,
    default: () => {
      return <>Boilerplate</>;
    },
  };
});

afterEach(cleanup);

describe("Welcome page", () => {
  it("renders the component", () => {
    const { container } = render(<Welcome />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("has no axe violations", async () => {
    const { container } = render(<Welcome />);
    expect(await axe(container)).toHaveNoViolations();
  });
});
