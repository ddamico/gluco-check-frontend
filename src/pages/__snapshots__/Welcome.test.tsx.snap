// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome page renders the component 1`] = `
<div
  class="MuiGrid-root makeStyles-contentGrid-4 MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-direction-xs-column MuiGrid-wrap-xs-nowrap MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
>
  <div
    class="MuiGrid-root makeStyles-subtitle-2 MuiGrid-item"
  >
    <div
      class="MuiContainer-root MuiContainer-maxWidthSm"
    >
      <h2
        class="MuiTypography-root MuiTypography-h5"
      >
        welcome.subtitle
      </h2>
    </div>
  </div>
  <div
    class="MuiGrid-root MuiGrid-item"
  >
    Onboarding
  </div>
  <div
    class="MuiGrid-root MuiGrid-item"
  >
    <p
      class="MuiTypography-root MuiTypography-body1"
    >
      <span>
        welcome.cta
      </span>
    </p>
  </div>
  <div
    class="MuiGrid-root makeStyles-boilerplate-5 MuiGrid-item"
  >
    Boilerplate
  </div>
</div>
`;
